src/FffPolygonGenerator.cpp

```diff
@@ -991,18 +991,29 @@ void FffPolygonGenerator::processPlatformAdhesion(SliceDataStorage& storage)
 {
     const Settings& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
     ExtruderTrain& train = mesh_group_settings.get<ExtruderTrain&>("adhesion_extruder_nr");
+
+    Polygons first_layer_outline;
+    coord_t primary_line_count;
     switch(mesh_group_settings.get<EPlatformAdhesion>("adhesion_type"))
     {
     case EPlatformAdhesion::SKIRT:
-        SkirtBrim::generate(storage, train.settings.get<coord_t>("skirt_gap"), train.settings.get<size_t>("skirt_line_count"));
+        primary_line_count = train.settings.get<size_t>("skirt_line_count");
+        SkirtBrim::getFirstLayerOutline(storage, primary_line_count, true, first_layer_outline);
```

Comment:
Why did you move this call here instead of from `generate()`? You could also get the line count from the `generate` function, right?

Reply:
We need the line count for the `getFirstLayerOutline` call as well. (Which was split off from `generate` because we can then differentiate between the case where the first layer outline is the entire plate versus just the tower.)